<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<meta name="description" content="MyWorld of programming from data to solution">
		<meta name="keywords" content="web design, html, css, javascript, python, R, SQL, big data, hadoop, spark, data sciences, data science, data analysis, data mining, machine learning, deep learning, artificial intelligence, al">
		<meta name="author" content="C.H.">
		<meta name="copyright" content="&copy; 2018 data sciences">
		<meta name="date" content="2018-01-10" scheme="YYYY-MM-DD">
		<title>Data Sciences|About</title>
		<link rel="stylesheet" href="./css/style.css">
		<link rel="shortcut icon" type="image/x-icon" href="img/mlp7.jpg" />
		<script type="text/javascript" src="https://latex.codecogs.com/latexit.js"></script>
		
		<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-116330579-1"></script>
		<script>
 		window.dataLayer = window.dataLayer || [];
  		function gtag(){dataLayer.push(arguments);}
 		gtag('js', new Date());
		gtag('config', 'UA-116330579-1');
		</script>		
 </head>
 <body>
		<header>
			<div class="container">
				<div id="branding">
					<h1><span class="highlight"><a class="branding" href="index.html">DATA SCIENCES</a></span></h1>
				</div>
				<nav class="navbar">
					<span class="open-slide">
						<a href="#" onclick="openSlideMenu()">
							<svg width="30" height="30">
								<path d="M0,5,30,5" stroke="#fff"
								stroke-width="5"/>
								<path d="M0,14,30,14" stroke="#fff"
								stroke-width="5"/>
								<path d="M0,23,30,23" stroke="#fff"
								stroke-width="5"/>
							</svg>
						</a>
					</span>
					<ul class="navbar-nav">
						<li><a href="index.html">Home</a></li>
						<li><a href="python.html">Python</a></li>
						<li><a href="r.html">R</a></li>
						<li><a href="sql.html">SQL</a></li>
						<li><a href="javascript.html">JavaScript</a></li>
						<li><a href="bigdata.html">Bigdata</a></li>
						<li class="current"><a href="about.html">About</a></li>
					</ul>
				</nav>

				<div id="side-menu" class="side-nav">
					<a href="#" class="btn-close" onclick="closeSlideMenu()">&times;</a>
					<a href="index.html">Home</a>
					<a href="python.html">Python</a>
					<a href="r.html">R</a>
					<a href="sql.html">SQL</a>
					<a href="javascript.html">JavaScript</a>
					<a href="bigdata.html">Bigdata</a>
					<a href="about.html">About</a>
				</div>
				
				<script>
					function openSlideMenu(){
						document.getElementById('side-menu').style.width
						='150px';
						document.getElementById('main').style.marginLeft
						='250px';
					}

					function closeSlideMenu(){
						document.getElementById('side-menu').style.width
						='0';
						document.getElementById('main').style.marginLeft
						='0';
					}
				</script>
			</div>
		</header>
		
		<section id="main">
			<div class="container">
				<aside id="sidebar">
					<br style ="line-height:4;">
					<div class="navy">
						<br>
						<p>
						<img src="aimg/aimg3a.jpg" alt="author" height="36%" width="100%" align="right">
						</p>
						<br>
						<p align="center">The author</p>
					</div>
					<br><br>
				</aside>
				<article id="main-col">
					<br>
					<h1 class="page-title">About the Author</h1>
					<p>
					<strong>Education</strong>
					<ul>
							<li><b>Ph.D.:</b> Plant Science, <a href="https://www.usask.ca/" target="_blank">University of Saskatchewan</a>, <a href="https://www.saskatoon.ca/" target="_blank">Saskatoon, Canada.</a></li>					
							<li><b>M.S.:</b> Statistical and Quantitative Genetics, <a href="http://www.njau.edu.cn/" target="_blank">Nanjing Agricultural University</a>, <a href="http://www.cityofnanjing.com/" target="_blank">Nanjing, China.</a></li>
							<li><b>B.S.:</b> Agronomy, <a href="http://www.hunau.edu.cn/" target="_blank">Hunan Agricultural University</a>, <a href="http://en.changsha.gov.cn/" target="_blank">Changsha, China.</a></li>
					</ul>
					</p><br>
					<p>
					<strong>The early experience in programming for his Master’s degree</strong>
					</p>
					<p>
					Many years ago, while pursuing his graduate studies for the Master’s Degree in Statistical and Quantitative Genetics at <a href="http://www.njau.edu.cn/" target="_blank">Nanjing Agricultural University</a>, the author had the opportunity to take the programming courses, i.e. Fortran (formerly FORTRAN from “FORmula TRANslation”) and BASIC (“Beginner's All-purpose Symbolic Instruction Code”). Soon, he created genetic algorithm using the programming languages to develop quantitative genetic models for data analysis and prediction of various genetic parameters for the quantitatively inherited traits of soybean for his M.S. thesis project. 
					</p>
					<p>
					By the way, Fortran is a general-purpose programming language which is well suited to numeric computation & scientific computing. It was developed in the early 1950s by the team led by John W. Backus at IBM and later released in 1957. However, it's a compiled language compared with other interpreted languages such as Python and JavaScript. BASIC is a general-purpose and high-level programming language which was designed by John G. Kemeny and Thomas E. Kurtz at Dartmouth College in New Hampshire, USA in 1964. 
					</p><br>
					<p>
					<strong>Computer simulation for his PhD project</strong>
					</p>
					<p>
					One of his earlier experience was the computer simulation of the distributions of the phenotypic data which were influenced by both genetic and environmental factors as part of the research proposal for his PhD project at the <a href="https://www.usask.ca/" target="_blank">University of Saskatchewan</a>. The phenotypic data, due to the genetic models and the modes of gene actions, depended predominantly on the number of genes as well as the environmental factors varied by the locations and years for the resistance to the common bunt disease of wheat using <a href="https://www.sas.com/en_us/home.html/" target="_blank">SAS</a> and <a href="https://www.minitab.com/en-us/" target="_blank">MINITAB</a> packages. Referring to the genetic approach of Robert C. Elston, the initial intention was to explore the possibility of grouping or clustering the <img src="https://latex.codecogs.com/gif.latex?F_{2}" title="F_{2}" />-derived progenies or families from the breeding populations to determine the number of genes controlling the disease resistance through the analysis of Mendelian inheritance, using simulated datasets based on the binomial distribution:
					<br><br>
					<img src="https://latex.codecogs.com/gif.latex?P_{p}(n|N)=\binom{N}{n}p^{n}q^{N-n}=\frac{N!}{n!(N-n)!}p^{n}(1-p)^{N-n}" title="P_{p}(n|N)=\binom{N}{n}p^{n}q^{N-n}=\frac{N!}{n!(N-n)!}p^{n}(1-p)^{N-n}" />
					<br><br>
					where <em>p</em> is the probability of common bunt disease for a wheat plant while <em>q</em> is the probability of being healthy for the plant (<em>q=1-p</em>), <em>n</em> is the number of diseased plants while <em>N</em> is the total number of wheat plants evaluated. The expected mean is <img src="https://latex.codecogs.com/gif.latex?\mu&space;=np" title="\mu =np" /> with a variance <img src="https://latex.codecogs.com/gif.latex?\sigma&space;^{2}=npq=np(1-p)" title="\sigma ^{2}=npq=np(1-p)" />.
					<br><br>
					Looking back the simulation with the concept of <em>machine learning (unsupervised learning)</em> in mind, the objective of the simulation was basically to draw inferences for classifying genotypic groups from the datasets simulated with the genetic models under different assumptions through the cluster analyses such as k-means, mixture models and hierarchical clustering.		
 					</p>
					<p>With his background and interest, he is inspired by the following quote from Dr. Robert C. Elston, a distinguished statistical geneticist and professor at <a href="https://case.edu/" target="_blank">Case Western Reserve University</a> since 1995, who was formerly a professor and head (1979-1995) in the Department of Biometry and Genetics at <a href="https://www.medschool.lsuhsc.edu/genetics/" target="_blank">Louisiana State University Medical Center</a> and a professor (prior to 1979) in the Department of Biostatistics at <a href="https://sph.unc.edu/bios/biostatistics/" target="_blank">University of North Carolina</a>:</p>
					<p>
					<em>"Statistical genetics may go out of fashion, but there will always be a need for statisticians who can compute."     -- Robert C. Elston (2015)<br>
					(Zheng et al. 2015. A Conversation with Robert C. Elston. Statistical Science 30(2) : 258–267. <a href="https://arxiv.org/abs/1507.08500"> DOI: 10.1214/14-STS497.</a>)</em>
					</p><br>
					<p>
					<strong>His recent roles with data sets</strong>
					</p>
					<p>
					Since 2015, Dr. He has been working as the Alfalfa Breeder's Toolbox (ABT) curator at <a href="https://www.noble.org/" target="_blank"> Noble Research Institute</a>. Some of his responsibilities in the ABT project relate to the <a href="https://www.alfalfatoolbox.org/jbrowse" target="_blank">genomic</a>, <a href="https://www.alfalfatoolbox.org/lbgo" target="_blank">genetic</a> and <a href="https://www.alfalfatoolbox.org/node/923831" target="_blank">phenotypic</a> datasets for their integration into the toolbox web portal to provide user-friendly access to the research community, for instance, to find the homologous/orthologous genes through sequence blasting against the reference genomes of the Cultivated Alfalfa at the Diploid Level (<a href="https://www.alfalfatoolbox.org/doblast/?filterword=DOBLAST&function=function2" target="_blank">CADL)</a> (<em>Medicago sativa</em>) and <a href="https://www.alfalfatoolbox.org/blast/nucleotide/nucleotide" target="_blank"><em>M. truncatula</em></a>, the model legume species. The ABT aims to provide resources for molecular breeding with efficiency to increase genetic gain not only for the forage crop but for the beef industry, ultimately.
					</p>
					<p>Previously, Dr. He worked for the CGIAR Generation Challenge Programme (<a href="http://www.generationcp.org/gcp-people" target="_blank">GCP</a>), currently the Integrated Breeding Platform (<a href="https://www.integratedbreeding.net/" target="_blank">IBP</a>), c/o <a href="https://www.cimmyt.org/" target="_blank">CIMMYT</a>, which was mainly funded by the <a href="https://www.gatesfoundation.org/" target="_blank">Bill & Melinda Gates Foundation (BMGF)</a>. Then, he managed and coordinated a lot of genotyping projects to assist many research programs in generating genotypic data for molecular marker assisted breeding of crop species to increase genetic gain in the developing countries across different continents. For instance, Dr. He, together with his colleagues, coordinated the development of many sets of <a href="https://www.integratedbreeding.net/482/communities/genomics-crop-info/crop-information/gcp-kaspar-snp-markers" target="_blank">SNP markers</a> and advocated their applications to target important genes of the 11 key crops by the scientific research communities. All sets of the KASP SNP markers were carefully selected including those from the important gene sequences of interest and developed across the genome of the relevant crop species. Many of them are associated with the economically important traits such as yield, quality, disease resistance and stress tolerance etc. He worked closely with the plant scientists to generate genomic, genetic and phenotypic datasets and conduct molecular breeding, some of his efforts are demonstrated through his introduction in the <a href="https://soundcloud.com/ibplatform/episode-5-english-interview" target="_blank">program interview</a> (including the <a href="https://soundcloud.com/ibplatform/u0zihaol7z8b" target="_blank">Chinese version</a>). Also, as he has continued introducing the <a href="https://link.springer.com/protocol/10.1007/978-1-4939-0446-4_7" target="_blank">KASP assay</a> through his publication for generating genotypic data, many research programs have adopted the cost-effective KASP assay into their own research laboratories.
					</p><br>
					
					<h1 class="page-title">About the Website</h1>
					<p>
					This site aims to host information on data mining, processing and analyses, data visualization and provide possible solutions to real world problems using statistics and computer programming.
					</p>	
				</article>
			</div>
			
			<div class="container">
				<article id="main-col">
					<h1 class="page-title">Contact the Author</h1>
					<p>
					If you have any questions, suggestions or comments, you may contact the author by <a href="mailto:chhe09@gmail.com">email</a> or by sending message and contact information below.
					</p>
					<form action="https://formspree.io/chhe09@gmail.com" method="POST">
					<p>Name:<br><input type="text" name="name" placeholder="Your name" size="39"></p>
					<p>Email:<br><input type="email" name="_replyto" placeholder="Your email" size="39"></p>
					<p>Message:<br><textarea name="message" placeholder="Your message" cols="37" rows="6"></textarea></p>
					<input type="submit" value="Send"><br>
					</form>
					<br>
				</article>
			</div>

			<div class="container">
				<article id="main-col">
					<h1 class="page-title"><small>Disclaimer</small></h1>
					<p><small>The contents on this site are for preliminary information purposes only based on data-related technologies such as data mining, processing, analyses, visualization as well as machine learning, etc.</small> 
					</p>
					<br>
				</article>
			</div>
	        </section>
		<br>
		
		<footer id="small">
			<p>Data Sciences, Copyright &copy; 2019,
			<a href="https://twitter.com/ChunlinHe" target="_blank"><img src="img/tweet_icon.png" width="31" height="31" alt="" /></a>
			<a href="https://www.linkedin.com/in/chunlinhe888" target="_blank"><img src="img/in_icon.png" width="31" height="31" alt="" /></a>
			</p> 
			<p id="time"></p>
			<script>
			document.getElementById("time").innerHTML = Date();
			</script>
		</footer>

</body>
</html> 
