<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<meta name="description" content="MyWorld of programming from data to solution">
		<meta name="keywords" content="web design, html, css, javascript, python, R, SQL, big data, hadoop, spark, data science, data analysis, data mining, machine learning, deep learning, artificial intelligence, al">
		<meta name="author" content="C.H.">
		<meta name="copyright" content="&copy; 2018 data sciences">
		<meta name="date" content="2018-01-10" scheme="YYYY-MM-DD">
		<title>Data Sciences|Python</title>
		<link rel="stylesheet" href="./css/style.css">
		<link rel="shortcut icon" type="image/x-icon" href="img/mlp7.jpg" />

		<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-116330579-1"></script>
		<script>
 		window.dataLayer = window.dataLayer || [];
  		function gtag(){dataLayer.push(arguments);}
 		gtag('js', new Date());
		gtag('config', 'UA-116330579-1');
		</script>
 </head>
 <body>
		<header>
			<div class="container">
				<div id="branding">
					<a href="index.html">
					<svg width="230" height="70">
					   <circle cx="32" cy="10" r="10" fill="#ff00d9"/>
					   <circle cx="10" cy="32" r="10" fill="#00FF00"/>
					   <circle cx="32" cy="32" r="10" fill="#00FFFF"/>
					   <circle cx="54" cy="32" r="10" fill="#ffff00"/>
					   <circle cx="32" cy="54" r="10" fill="#ff0000"/>
					   <text x="70" y="40" font-size="20px" font-weight="bold" fill="#00FF00">CHUNLIN HE</text>
					</svg>
					</a>
					<!-- <h1><span class="highlight"><a class="branding" href="index.html">DATA SCIENCES
					</a></span>
					</h1> -->
				</div>

				<nav class="navbar">
					<span class="open-slide">
						<a href="#" onclick="openSlideMenu()">
							<svg width="30" height="30">
								<path d="M0,5,30,5" stroke="#fff"
								stroke-width="5"/>
								<path d="M0,14,30,14" stroke="#fff"
								stroke-width="5"/>
								<path d="M0,23,30,23" stroke="#fff"
								stroke-width="5"/>
							</svg>
						</a>
					</span>
					<ul class="navbar-nav">
						<li><a href="index.html">Home</a></li>
						<li class="current"><a href="python.html">Python</a></li>
						<li><a href="r.html">R</a></li>
						<li><a href="sql.html">SQL</a></li>
						<li><a href="javascript.html">JavaScript</a></li>
						<li><a href="bigdata.html">Bigdata</a></li>
						<li><a href="about.html">About</a></li>
					</ul>
				</nav>

				<div id="side-menu" class="side-nav">
					<a href="#" index.html="btn-close" onclick="closeSlideMenu()" align="right">Close &times;</a>
					<a href="index.html">Home</a>
					<a href="python.html">Python</a>
					<a href="r.html">R</a>
					<a href="sql.html">SQL</a>
					<a href="javascript.html">JavaScript</a>
					<a href="bigdata.html">Bigdata</a>
					<a href="about.html">About</a>
				</div>

				<script>
					function openSlideMenu(){
						document.getElementById('side-menu').style.width
						='150px';
						document.getElementById('main').style.marginLeft
						='250px';
					}

					function closeSlideMenu(){
						document.getElementById('side-menu').style.width
						='0';
						document.getElementById('main').style.marginLeft
						='0';
					}
				</script>
			</div>
		</header>

		<section id="main">
			<div class="container">
				<aside id="sidebar">
					<br style ="line-height:4;">
					<div class="navy">
						<img src="pimg/pimg3_logo.png" alt="p-icon" height="100" width="240" align="middle">
						<h3>Why Python?</h3>
						<p>Python is an easy to learn, open source and object-oriented programming language. It has a great number of libraries and tools which make the tasks easy. This website aims to use Python for data analysis in general and to learn more about its capabilities for computational analysis.  As new high-throughput experimental techniques with cost-effective data generation on the rise, data analysis capabilities are sought-after features for research programs.</p>
					</div>
				</aside>

				<article id="main-col">
					<br>
					<h1 class="page-title">Python Programming</h1>
					<h3>Python basics</h3>
					<p>
					Python, developed by Guido van Rossum, is not only a free software but also a powerful modern computer programming language. Python is a high-level but easy-to-learn interactive, interpreted, and object-oriented scripting language. Python is highly readable since the English keywords are frequently used instead of punctuations used by other languages, and it has fewer syntactical constructions than other languages.
					</p>
					<p>
					There are some similarities between Python and Fortran, one of the earliest programming languages, but Python is much more powerful than Fortran. Python uses variables without declaring them and uses indentations as a control structure. Unlike Java, programmers don’t have to define classes in Python but are free to do so when necessary.
					</p>
					<p>
					In addition, Python is also a major tool for web development.
					</p>
					<h3>Data analysis</h3>
					<p>
					Python has grown and got dedicated open source libraries substantially from the community over the last several years for data analysis and predictive modeling through its evolution from originally a general purpose language. This includes the fundamental packages and libraries for scientific computing, data analysis and statistical applications such as NumPy, IPython, matplotlib, and pandas etc.
					</p>
					<h3>Data visualization</h3>
					<p>
					What’s called good data means the data that one can easily visualize and understand. In data science, generating the most comprehensive and understandable visualizations is highly desirable. Datasets and their sources vary in types including remote data, local data, JSON, CSV etc as well as the data from relational databases. However, the process is not always simple since one has to find the data, process them by reading, cleaning and massaging, and then use the right tool to visualize them.
					</p>
					<p>
					Basic simple plots could easily be created in Python using matplotlib, however, the more advanced charting for the most appealing visualizations requires, and can only be accomplished with the knowledge and skills of Python in combination with other tools.
					</p>
					<h3>Machine learning and artificial intelligence</h3>
					<p>
					Python is known for its simplicity and thus considered to be an ideal language for machine learning (ML) and artificial intelligence (AI) (<a href="https://amzn.to/2TLEIa9" target="_blank">Artificial Intelligence with Python, 2017</a>). The library, Pybrain, is especially rising for machine learning in Python.
					</p>
				</article>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<br>
					<h1 id="libraries2018">The Git Statistics of Python Libraries (2018)</h1>
					<p>The statistics has come out on the most important python libraries for data science in 2018 summarized from the github site as shown in the <a href="https://www.datasciencecentral.com/profiles/blogs/top-20-python-libraries-for-data-science-in-2018" target="_blank">Data Science Central</a> and <a href="https://activewizards.com/blog/top-20-python-libraries-for-data-science-in-2018/" target="_blank">ActiveWizards</a>. Among them, TensorFlow stands out with the most number of commits, stars and forks.</p>
					<p>A list of the top 20 python libraries with detailed activities from git in 2018:<br>
					<img src="pimg/pimg9.jpg" alt="libraries" align="middle" width="100%">
					</p>
					<br>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<img src="pimg/pimg10b.jpg" alt="p-frameworks" height="300%" width="100%">
						<br>
						<p>No. 1 python package with the top git activities in 2018.</p>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h1 id="frameworks">The Popular Frameworks of Python (2017)</h1>
					<p>Based on the developers’ survey in 2017, Django is the most popular framework of Python used by 41% of developers while NumPy, Pandas, Matplotlib are the second popular frameworks all together used by 39% of the developers.</p>
					<p>A list of the top 15 python libraries with detailed activities from git in 2017:<br>
					<img src="pimg/pimg5_top15libs.jpg" alt="libraries" align="middle" width="100%">
					</p>
					<br>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<a href="pimg/pimg4_framework.jpg" style="text-decoration:none;">
						<img src="pimg/pimg4_framework.jpg" alt="p-frameworks" height="300%" width="100%">
						<br>
						<p><font color="white">Figure is adapted from Python Software Foundation & JetBrains.</font></p>
						</a>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">

			<div class="container">
				<article id="main-col">
					<hr><br>

					<h1>Python Applications</h1>

					<h3 id="covid-19">The prediction of covid-19 pandemic</h3>
					<p><b>The pandemic:</b> the covid-19 pandemic was caused by the pneumonia-causing novel coronavirus (<a href="https://www.cdc.gov/coronavirus/2019-ncov/lab/grows-virus-cell-culture.html" target="_blank">SARS-CoV-2</a>), which has caused lethal respiratory infections in humans in 2020. The coronaviruses may also cause many diseases in animals such as cows, chicken, pigs and birds. The outbreak was believed to be initiated from either a single introduction into humans or very few animal-to-human transmissions (Johns Hopkins Center for Health Security, April 2020). </p>
					<p><b>The viral genome (GenBank Accession: <a href="https://www.ncbi.nlm.nih.gov/nuccore/1798174254" target="_blank">NC_045512</a>):</b> the covid-19 is positive-sense RNA virus and has approximately 25-32 kilobases being the largest genome of RNA viruses with a unique replication capability. Genomic research indicated that two genes, i.e. the S and N genes, are involved in the SARS-CoV-2. High mutation rates of the RNA viruses may result in several slightly different versions of the viral genome whenever the viral genome is replicated, thus creating a new viral population with diverse genomes or, namely, viral quasispecies. </p>
					<p>NEXTSTRAIN created a chart below to illustrate the viral genome structural diversity and a phylogeny generated from the genome sequence analysis of samples collected from different countries (nextstrain.org, May 2020).</p>
					<b>Diversity of the viral genomes including the open reading frames (ORFs):</b>
					<a href="pimg/pimg26.png" style="text-decoration:none;">
					<img src="pimg/pimg26.png" alt="diversity" align="middle" width="100%"></a>
					<br><br>
					<b>A phylogeny from the genome sequence analysis of the (+)ssRNA viruses from many countries:</b>
					<a href="pimg/pimg27.jpg" style="text-decoration:none;">
					<img src="pimg/pimg27.jpg" alt="phylogeny" align="middle" width="100%"></a>
					<br>
					<p>The genome sequence analysis across samples has demonstrated highest diversity occurring in the structural genes, especially the S protein, ORF3a, and ORF8 (Johns Hopkins Center for Health Security, April 2020).</p>
					<p>Another study has reported that selective pressure on the virus may have resulted in mutation based on the analysis of Open Reading Frame 1ab (ORF1ab) of COVID‐2019 and the stabilizing mutation at nsp2 protein has made the COVID‐19 disease being more contagious than SARS  (Angeletti et al., J. Med. Virol., Feb. 2020). </p>
					<p><b>The prediction of covid-19 pandemic:</b> the data were obtained and derived from the Johns Hopkins University and the pandemic prediction was made through the use of a python package, i.e. Prophet.  The data visualization and the charts of predictions on the cumulative cases as well as the new cases for some of the States (i.e. <a href="pimg/pimg22.jpg" target="_blank">WA</a>, <a href="pimg/pimg21.jpg" target="_blank">CA</a>, <a href="pimg/pimg19.jpg" target="_blank">IN</a>, <a href="pimg/pimg20.jpg" target="_blank">NY</a>) and <a href="pimg/pimg18.jpg" target="_blank">Hubei</a> province are illustrated in the right column.
					Also, animated data visualizations are created for <a href="index.html#cumulative_cases" target="_blank">cumulative</a> cases and <a href="index.html#new_cases" target="_blank">new</a> cases across the states.
					</p>

					Another example in controlling the covid-19 pandemic in the same chart for WA, CA, IN, NY:
					<a href="pimg/pimg23.jpg" style="text-decoration:none;" target="_blank">
					<img id="covid19f" src="pimg/pimg23.jpg" alt="covid-19f" height="60%" width="100%"></a>

					<p><b>Containing the pandemic:</b> by August 17, 2020, the countries that have more than half million confirmed covid-19 cases include USA, Brazil, India, Russia, South Africa, Peru and Mexico based on the <a href="pimg/pimg28.mp4" target="_blank">cumulative</a> cases and the <a href="pimg/pimg29.mp4" target="_blank">new</a> cases (Data from Johns Hopkins University, August 2020).
					In the USA, by August 23, 2020, the states having more than 150k total confirmed covid-19 cases include California, Florida, Texas, New York, Georgia, Illinois, Arizona, New Jersey and North Carolina as shown in the figures below, including the animated charts for the <a href="pimg/pimg30.mp4" target="_blank">cumulative</a> cases and <a href="pimg/pimg31.mp4" target="_blank">new</a> cases.

					<a href="pimg/pimg32.jpg" style="text-decoration:none;" target="_blank">
					<img src="pimg/pimg32.jpg" alt="states" height="60%" width="100%"></a>

					<a href="pimg/pimg33.jpg" style="text-decoration:none;" target="_blank">
					<img src="pimg/pimg33.jpg" alt="states" height="60%" width="100%"></a>
					</p>

					<p><b>Testing for COVID-19:</b> there are generally two types of testing regarding the covid-19 infections including (1) diagnostic tests and (2) antibody tests.
					<ul>
						<li>Diagnostic testing: there are two types of diagnostic tests to detect the virus including (i) molecular tests or RT-PCR tests and (ii) antigen tests.</li>
							<ul>
								<li>Molecular tests: this is the genomic detection-based method to detect genetic material of the virus using the PCR technology, thus it’s also called a PCR test, nucleic acid amplification test (NAAT) or,
								simply, gene test. It is done by collecting fluid from a nasal, throat swab or saliva. Then, a pair of <a href="https://www.cdc.gov/coronavirus/2019-ncov/lab/rt-pcr-panel-primer-probes.html" target="_blank">PCR primers</a>, for instance, Forward Primer (GAC CCC AAA ATC AGC GAA AT) and Reverse Primer
								(TCT GGT TAC TGC CAG TTG AAT CTG), are used to generate an amplicon from the viral genome which is further detected by the machine using fluorescence marker. This method is able to detect active coronavirus infection.</li>
								<li>Antigen tests</li>
							</ul>
						<li>Antibody testing: This is the test to detect the presence of antibodies that are produced by the immune system to fight infections in response to the virus. It may take several days or weeks to develop antibodies after the infection which could stay in the blood for several weeks or more after recovery. Therefore, we should not use antibody tests to diagnose an active covid-19 infection.</li>
					</ul>
					</p>

				</article>

				<aside id="sidebar"><br style ="line-height:6;">
<!--				<div class="purple", id="covid19f">
					<a href="pimg/pimg23.jpg" style="text-decoration:none;">
					<img src="pimg/pimg23.jpg" alt="covid-19f" height="100%" width="100%">
					<font color="#ffffff">Controlling the covid-19 pandemic in WA, CA, IN, NY.</font>
					</a>
					</div>
-->
					<br style ="line-height:6;">
					<div class="purple", id="covid19e">
					<a href="pimg/pimg22.jpg" style="text-decoration:none;">
					<img src="pimg/pimg22.jpg" alt="covid-19e" height="100%" width="100%">
					<font color="#ffffff">Controlling the covid-19 pandemic in Washington (Courtesy: data from Johns Hopkins University).</font>
					</a>
					</div>

					<br style ="line-height:6;">
					<div class="purple", id="covid19d">
					<a href="pimg/pimg21.jpg" style="text-decoration:none;">
					<img src="pimg/pimg21.jpg" alt="covid-19d" height="100%" width="100%">
					<font color="#ffffff">Controlling the covid-19 pandemic in California (Courtesy: data from Johns Hopkins University).</font>
					</a>
					</div>

					<br style ="line-height:6;">
					<div class="purple", id="covid19c">
					<a href="pimg/pimg20.jpg" style="text-decoration:none;">
					<img src="pimg/pimg20.jpg" alt="covid-19c" height="100%" width="100%">
					<font color="#ffffff">Controlling the covid-19 pandemic in New York (Courtesy: data from Johns Hopkins University).</font>
					</a>
					</div>

					<br style ="line-height:6;">
					<div class="purple", id="covid19b">
					<a href="pimg/pimg19.jpg" style="text-decoration:none;">
					<img src="pimg/pimg19.jpg" alt="covid-19b" height="100%" width="100%">
					<font color="#ffffff">Controlling the covid-19 pandemic in Indiana (Courtesy: data from Johns Hopkins University).</font>
					</a>
					</div>

					<br style ="line-height:6;">
					<div class="purple", id="covid19a">
					<a href="pimg/pimg18.jpg" style="text-decoration:none;">
					<img src="pimg/pimg18.jpg" alt="covid-19a" height="100%" width="100%">
					<font color="#ffffff">Controlling the covid-19 pandemic in Hubei (Courtesy: data from Johns Hopkins University).</font>
					</a>
					</div>
				</aside>
			</div>

			<br>
			<div class="container">
				<article id="main-col">
					<h3 id="Text mining">Text mining analytics</h3>
					<p>To quickly understand the content of a document or webpage, we can use text mining to identify words and make a word cloud to visualize the key words with high frequency. A word cloud can intuitively present the results. It can be made using <a href="r.html#wordcloud2" style="text-decoration:none;">R</a> or Python. In python, the WordCloud package is used for data visualization for representing text data where the size of the word indicates its frequency with relative importance.</p>
					<p>In the meantime, other Python packages including matplotlib, numpy and PIL are used for the text data visualization. The Python WordCloud package can also allow user to adopt other shape to generate the words in that shape.  </p>
					<p>However, the WordCloud package does not support Chinese characters, thus it is required to add an argument for Chinese font type to support that function using “font_path” parameter. </p>
				</article>

				<aside id="sidebar"><br style ="line-height:6;">
					<div class="purple", id="wordcloud3">
					<a href="pimg/pimg17.jpg" style="text-decoration:none;">
					<img src="pimg/pimg17.jpg" alt="wordcloud" height="100%" width="100%">
					<font color="#ffffff">A word cloud generated through text mining using python.</font>
					</a>
					</div>
				</aside>
			</div>

			<div class="container">
				<article id="main-col">
					<br>
					<h3 id="static-bubble">Create a bubble chart</h3>
					<p>A bubble plot is similarly a scatterplot of x- and y-axis with additional third variable or dimension equivalent to a z-axis. The additional third numeric variable is indicated using the size of the bubbles. For example, in order to explore the dynamics of GMO soybean production (Herbicide-tolerant) in 2018, the GMO% of all soybeans planted is plotted for the states that produce most soybeans relative to their GDP where the bubble size is presented as the corresponding population, proportionally. Likewise, bubble charts are also created for the GMO% of all corn planted and the GMO% of all upland cotton planted across the major production States in 2018 that includes cultivars with stacked genes, herbicide-tolerance and insect-resistance/Bt.</p>

					<pre><code>
 					import matplotlib.pyplot as plt
					import numpy
					x, y, s = numpy.loadtxt('/mydata.csv', delimiter=',', unpack=True, usecols=(3, 4, 5), skiprows=1)
					GDP, GMO, pop = x, y, s
					colors = ['#002387', '#043927', '#4C2882', '#48BF91', '#4F42B5', '#800080', '#DA70D6', '#CFB53B', '#D1E231', '#D7837F', '#DF00FF', '#39FF14', '#FE2712', '#FF5800']
					state = ['AR', 'IL', 'IN', 'IA', 'KS', 'MI', 'MN', 'MS', 'MO', 'NE', 'ND', 'OH', 'SD', 'WI']
					plt.title('Bubble Chart for GMO% of All Soybeans Planted Across States (Herbicide-tolerant, 2018; bubble size corresponds to population)', color='#006400')
					plt.xlabel('State GDP (x1000M USD)', color='#006400')
					plt.ylabel('GMO% of All Soybeans Planted', color='#006400')
					plt.scatter(x=GDP, y=GMO, s=pop, alpha=0.6, c=colors)
					for i, txt in enumerate(state):
					    plt.annotate(txt, (x[i], y[i]), ha='center', va='center')
					plt.show()
					</code></pre>
				</article>

				<aside id="sidebar"><br style ="line-height:6;">
					<div class="purple">
					<a href="pimg/pimg16_cottonGMO.png" style="text-decoration:none;">
					<img src="pimg/pimg16_cottonGMO.png" alt="static-bubble" height="100%" width="100%">
					<font color="#ffffff">A bubble chart for GMO% of all upland cotton planted across USA in 2018.</font>
					</a>
					</div>

					<div class="purple">
					<a href="pimg/pimg15.png" style="text-decoration:none;">
					<img src="pimg/pimg15.png" alt="static-bubble" height="100%" width="100%">
					<font color="#ffffff">A bubble chart for GMO% of all corn planted across USA in 2018.</font>
					</a>
					</div>

					<div class="purple">
					<a href="pimg/pimg14.png" style="text-decoration:none;">
					<img src="pimg/pimg14.png" alt="static-bubble" height="100%" width="100%">
					<font color="#ffffff">A bubble chart for GMO% of all soybeans planted across USA in 2018.</font>
					</a>
					</div>
				</aside>
			</div>
			<br>

			<div class="container">
				<article id="main-col">
					<h3 id="rna-seq">Visualization of gene expression data</h3>
					<p>To follow the RNA-seq data analysis using <a href="r.html#rnacim">R</a> based on the hypothetical example from maize gene expression, a scatter plot matrix is created for the selected sample-treatment combinations using Python, where the diagonal displays the plots of the probability density function through kernel density estimation (KDE). </p>
					<pre><code>
					import pandas as pd
					data=pd.read_csv("transmydata.csv")
					import numpy as np
					import matplotlib.pyplot as plt
					from pandas.plotting import scatter_matrix
					df = pd.DataFrame(data, columns = ['CAW', 'CBW', 'CAD', 'CBD'])
					scatter_matrix(df, diagonal = ‘kde’, alpha = 0.6, figsize = (5, 5), c=df['CAD'])
					plt.show()
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="pimg/pimg13_rnaseq.jpg" style="text-decoration:none;">
					<img src="pimg/pimg13_rnaseq.jpg" alt="rna-seq" height="100%" width="100%">
					<font color="#ffffff">A scatter plot matrix for maize gene expression. The diagonal displays the probability density function using KDE.</font>
					</a>
					</div>
				</aside>
			</div>
			<div class="container">
				<article id="main-col">
					<h3 id="genomediagram">Genome diagrams</h3>
					<p>Visualization of syntenic regions across different species/subspecies or different strains of the same species using genome diagrams:</p>
					<pre><code>
					from Bio import SeqIO
					Seq1_data = SeqIO.read("seq_species1.gb", "gb")
					Seq2_data = SeqIO.read("seq_species2.gb", "gb")
					from Bio.Graphics import GenomeDiagram
					from reportlab.lib.colors import *
					Seq1_colors = [green]*5 + [blue]*7 + [orange]*2 + [grey]*2 + [red] + [lightblue]*11 + [green]*4 \
					         + [grey] + [green]*2 + [grey, green] + [brown]*5 + [blue]*4 + [purple]*5 \
					         + [grey, lightblue] + [navy]*2 + [grey]
					Seq2_colors = [green]*6 + [blue]*8 + [orange]*2 + [grey] + [red] + [lightblue]*21 + [green]*5 \
					         + [grey] + [brown]*4 + [blue]*3 + [purple]*3 + [grey]*5 + [navy]*2
					name = "SyntenyView_linear"
					gd_diagram = GenomeDiagram.Diagram(name) #to create genome diagrams
					max_len = 0
					for record, gene_colors in zip([Seq1_data, Seq2_data], [Seq1_colors, Seq2_colors]):
					    max_len = max(max_len, len(record))
					    gd_track_for_features = gd_diagram.new_track(1, start=0, end=len(record))
					    #can include (name="name of species/subspecies") to label the track.
					    gd_feature_set = gd_track_for_features.new_set()
					    i = 0
					    for feature in record.features:
					        if feature.type != "gene":
					            #Exclude current feature
					            continue
					        gd_feature_set.add_feature(feature, sigil="ARROW",
					                                   color=gene_colors[i], label=True,
					                                   name = str(i+1),
					                                   label_position="start",
					                                   label_size = 9, label_angle=90, label_color=green)
					        i+=1
					gd_diagram.draw(format="linear", pagesize='A2', fragments=1, start=0, end=max_len)
					    #or use format="circular" to create a circular graph.
					gd_diagram.write(name + ".pdf", "PDF")
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="pimg/pimg8.png" style="text-decoration:none;">
					<img src="pimg/pimg8.png" alt="linear" height="100%" width="100%">
					</a>
					<br>
					<a href="pimg/pimg12.png" style="text-decoration:none;">
					<img src="pimg/pimg12.png" alt="circular" height="100%" width="100%">
					<p><font color="#ffffff">Visualization of syntenic regions across the genomes of different species or subspecies using linear (top) and circular (bottom) diagrams.</font></p>
					</a>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h3>Generate a list of random primer sequences with GC content for <em>in silico</em> PCR</h3>
 					<pre><code>
 					import random
					from Bio.SeqUtils import GC
					for i in range(18):
					        def random_dna_seq_f(length):
					             return ''.join(random.choice('ACTG') for _ in range(length));
					        seq_f=random_dna_seq_f(20)
					        def random_dna_seq_r(length):
					             return ''.join(random.choice('ACTG') for _ in range(length));
					        seq_r=random_dna_seq_r(20)
					        print("Marker"+"{:3d}".format(i+1), ":", "F-primer:", seq_f, "GC%=",GC(seq_f), " ", "R-primer:", seq_r, "GC%=", GC(seq_r));
 					</code></pre>
 					<br>
					<img src="pimg/pimg6_primers.jpg" alt="p-primer" height="300%" width="100%">
				</article>
			</div>
		</section>
		<br>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h3 id="3dscatterplot">3D scatter plot of crop variety trials</h3>
					<pre><code>
					from mpl_toolkits.mplot3d import Axes3D
					import matplotlib.pyplot as plt

					fig=plt.figure() #create a figure object.
					ax=fig.add_subplot(111, projection='3d')

					X=[3,8.5,6,8,1,9,2,8,6,5,9,9]
					Y=[2,9,1,2,3,7,8,5,9,9,8,9]
					Z=[402,606,444,398,472,458,602,376,438,584,572,600]

					ax.scatter(X, Y, Z, c='#0000FF')
					ax.set_xlabel('Quality (9=Best)',color='#008000')
					ax.set_ylabel('Disease Resistance (9=Best)',color='#008000')
					ax.set_zlabel('Yield (kg/ac)',color='#008000')
					ax.set_title('3D SCATTER PLOT OF 12 CULTIVARS',color='#000080')
					plt.show()
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<a href="pimg/pimg2.jpg" style="text-decoration:none;">
						<img src="pimg/pimg2.jpg" alt="3D Scatter plot" height="300%" width="100%">
						</a>
					</div>
				</aside>
			</div>
			<div class="container">
				<article id="main-col">
					<h3 id="s3dplot">Simple 3D plot</h3>
					<pre><code>
					from mpl_toolkits.mplot3d import Axes3D
					import numpy as np
					import matplotlib.pyplot as plt
					fig=plt.figure(figsize=(10,10))
					ax=fig.add_subplot(111,projection='3d')

					u=np.linspace(0, 2*np.pi, 100)
					v=np.linspace(0, np.pi, 100)

					x=10*np.outer(np.cos(u),np.sin(v))
					y=10*np.outer(np.sin(u),np.sin(v))
					z=10*np.outer(np.ones(np.size(u)),np.cos(v))
					ax.plot_surface(x,y,z,rstride=4,cstride=4,color='#ADFF2F')
					ax.set_xlabel('X',color='#008000')
					ax.set_ylabel('Y',color='#008000')
					ax.set_zlabel('Z',color='#008000')
					ax.set_title('3D Surface Plot',color='#000080')
					plt.show()
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<a href="pimg/pimg1.jpg" style="text-decoration:none;">
						<img src="pimg/pimg1.jpg" alt="Simple 3D plot" height="300%" width="100%">
						</a>
					</div>
				</aside>
			</div>
		</section>
		<br><br>


		<footer>
			<p>Data Science, Copyright &copy; 2020
			<a href="https://twitter.com/ChunlinHe" target="_blank"><img src="img/tweet_icon.png" width="31" height="31" alt="" /></a>
			<a href="https://www.linkedin.com/in/chunlinhe888" target="_blank"><img src="img/in_icon.png" width="31" height="31" alt="" /></a>
			</p>
			<p id="time"></p>
			<script>
			document.getElementById("time").innerHTML = Date();
			</script>
		</footer>
</body>

</html>
