<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<meta name="description" content="MyWorld of programming from data to solution">
		<meta name="keywords" content="web design, html, css, javascript, python, R, SQL, big data, hadoop, spark, data sciences, data science, data analysis, data mining, machine learning, deep learning, artificial intelligence, al">
		<meta name="author" content="C.H.">
		<meta name="copyright" content="&copy; 2018 data sciences">
		<meta name="date" content="2018-01-10" scheme="YYYY-MM-DD">
		<title>Data Sciences|R</title>
		<link rel="stylesheet" href="./css/style.css">
		<link rel="shortcut icon" type="image/x-icon" href="img/mlp7.jpg" />

		<!-- Global site tag (gtag.js) - Google Analytics -->
		<script async src="https://www.googletagmanager.com/gtag/js?id=UA-116330579-1"></script>
		<script>
 		window.dataLayer = window.dataLayer || [];
  		function gtag(){dataLayer.push(arguments);}
 		gtag('js', new Date());
		gtag('config', 'UA-116330579-1');
		</script>
 </head>
 <body>
		<header>
			<div class="container">
				<div id="branding">
					<a href="index.html">
					<svg width="230" height="70">
					   <circle cx="32" cy="10" r="10" fill="#ff00d9"/>
					   <circle cx="10" cy="32" r="10" fill="#00FF00"/>
					   <circle cx="32" cy="32" r="10" fill="#00FFFF"/>
					   <circle cx="54" cy="32" r="10" fill="#ffff00"/>
					   <circle cx="32" cy="54" r="10" fill="#ff0000"/>
					   <text x="70" y="40" font-size="20px" font-weight="bold" fill="#00FF00">CHUNLIN HE</text>
					</svg>
					</a>
					<!-- <h1><span class="highlight"><a class="branding" href="index.html">DATA SCIENCES
					</a></span>
					</h1> -->
				</div>

				<nav class="navbar">
					<span class="open-slide">
						<a href="#" onclick="openSlideMenu()">
							<svg width="30" height="30">
								<path d="M0,5,30,5" stroke="#fff"
								stroke-width="5"/>
								<path d="M0,14,30,14" stroke="#fff"
								stroke-width="5"/>
								<path d="M0,23,30,23" stroke="#fff"
								stroke-width="5"/>
							</svg>
						</a>
					</span>
					<ul class="navbar-nav">
						<li><a href="index.html">Home</a></li>
						<li><a href="python.html">Python</a></li>
						<li class="current"><a href="r.html">R</a></li>
						<li><a href="sql.html">SQL</a></li>
						<li><a href="javascript.html">JavaScript</a></li>
						<li><a href="bigdata.html">Bigdata</a></li>
						<li><a href="about.html">About</a></li>
					</ul>
				</nav>

				<div id="side-menu" class="side-nav">
					<a href="#" index.html="btn-close" onclick="closeSlideMenu()" align="right">Close &times;</a>
					<a href="index.html">Home</a>
					<a href="python.html">Python</a>
					<a href="r.html">R</a>
					<a href="sql.html">SQL</a>
					<a href="javascript.html">JavaScript</a>
					<a href="bigdata.html">Bigdata</a>
					<a href="about.html">About</a>
				</div>

				<script>
					function openSlideMenu(){
						document.getElementById('side-menu').style.width
						='150px';
						document.getElementById('main').style.marginLeft
						='250px';
					}

					function closeSlideMenu(){
						document.getElementById('side-menu').style.width
						='0';
						document.getElementById('main').style.marginLeft
						='0';
					}
				</script>
			</div>
		</header>

		<section id="main">
			<div class="container">
				<aside id="sidebar">
					<br style ="line-height:4;">
					<div class="navy">
						<img src="rimg/rimg2.png" alt="r-icon" height="32%" width="32%" align="right">
						<h3>Why R?</h3>
						<p>R has become a popular programming language for data analysis, visualization, and statistical modeling as well as for machine learning.  As an open source programming language and environment, R aims to provide information for wet-lab researchers who want to use R for data processing, and for bioinformaticians who are new to R but want to learn it.</p>
					</div>
				</aside>
				<article id="main-col">
					<br>
					<h1 class="page-title">R Programming</h1>
					<h3>R basics</h3>
					<p>
					R is a free programming language and software environment for statistical computing and graphics that is supported by the R Foundation for Statistical Computing. The R language is widely used among statisticians and data miners for developing statistical software and data analysis.
					</p>
					<p>
					This is simply an introduction on using R for computational genomics. It is aimed at wet-lab researchers who wants to use R in their data analysis ,and bioinformaticians who are new to R and wants to learn more about its capabilities for genomics data analysis. As new high-throughout experimental techniques on the rise, data analysis capabilities are sought-after features for researchers. R, with its statistical heritage, plotting features and rich user-contributed packages is one of the best languages for the task of analyzing data.
					</p>
					<p>
					R has become more and more important software language for carrying out both simple and complicated statistical analyses, including:
					</p>
					<ul>
							<li>Data summary and exploration</li>
							<li>Graphical presentation</li>
							<li>Data modelling</li>
					</ul>
					<p>
					R is considered as one of the most effective environment and language for manipulating and analyzing data in statistics. It can generate graphics with high quality for publication including relevant formulae and mathematical symbols.
					</p>
					<p>
					Unlike other statistical softwares, R stores information and operates on objects, such as scalars, vectors, matrices, lists and dataframes etc. It is very useful to define new types of object to target particular applications.  One of R's essential features is that it will do different things for different types of objects.
					</p>
					<h3>Machine learning and artificial intelligence</h3>
					<p>
					Apart from being one of the important general purpose languages, R also has many packages such as Gmodels, RODBC, Class and Tm that can be used for machine learning and artificial intelligence. The packages can facilitate and implement the machine learning algorithms solve the business oriented problems (<a href="https://amzn.to/2SRxW5g" target="_blank">Machine Learning with R, 2017</a>).
					</p>
				</article>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h1 id="rlibraries">The Git Statistics of the Most Popular R Libraries (2018)</h1>
					<p>Like Python, R is another important and popular tool for data sciences that is mainly designed for statistical computing.  R has been well developed with thousands of packages to solve a lot of problems in data analysis. Listed below are the most important R libraries and packages in 2018 summarized from the github site by the <a href="https://activewizards.com/blog/top-20-r-libraries-for-data-science-in-2018-infographic/" target="_blank">ActiveWizards</a>. The most popular libraries include dplyr, ggplot2, plotly, knitr and XGBoost based on the number of commits and contributors suited for data manipulation, visualization, competitive research as well as machine learning.
					</p>
					<p>A list of the top 20 R libraries from GitHub:<br>
					<img src="rimg/rimg15.jpg" alt="rlibraries" align="middle" width="100%">
					</p>
					<br>
					<hr>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<a href="rimg/rimg16_lib.jpg" style="text-decoration:none;">
						<img src="rimg/rimg16_lib.jpg" alt="rlibraries" height="100%" width="100%">
						<br>
						<p><font color="white">The most popular R libraries (2018).</font></p>
						</a>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<br>
					<h1>R Applications</h1>
					<br>
					<h3 id="inbredhybrid">Text mining analytics</h3>
					<p>Text mining analytics is used to identify and highlight the most frequently used keywords from an existing environment such as article or a web page.  It aims to process raw information to bring out meaningful insights. </p>
					<p>A word cloud, and the like, is a graphical display of frequently used words in a collection of text derived from documents or websites, which demonstrates the frequency of words by their font sizes for simple visualization. It is ideal for a quick and qualitative view of information to grasp attention.  Thus, it has its advantages of being (1) clarity on what stands out, (2) communicative tool with impact, and (3) a simple art with special visual effects.</p>
					<p>Users of a word cloud are across many different fields including data scientists for quick visualization and reports of qualitative data.  A word cloud can be generated using some of the R packages including a text mining package (tm) and a generator package (wordcloud).  </p>

				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="rimg/rimg17.jpg" style="text-decoration:none;">
					<img src="rimg/rimg17.jpg" alt="wordcloud" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">Word cloud through text mining using R </font></p>
					</a>
					</div>
				</aside>
			</div>


			<div class="container">
				<article id="main-col">
					<br>
					<h3 id="rnacim">RNA-seq data analysis</h3>
					<p>Since the availability of the next-generation sequencing (NGS) technology, RNA-seq has become popular in biological research in many species for gene discovery, annotation and expression analysis along with the identification of SNP markers across a genome. Compared with the previous DNA microarray technology, it can provide a comprehensive overview of transcriptome with some of the great advantages including higher sensitivity, higher resolution at nucleotide level and no prerequisite of gene sequences.  The advances in high-throughput sequencing with decreasing cost have made it affordable to rapidly generate a large quantity of DNA sequences and genotypic data for investigating a wide range of biological and medical problems.     In the meantime, massive and complex sequence datasets generated from the sequencers create a need for the development of statistical and bio-computational applications to tackle the analysis and data management.
					</p>
					<p id="rnahm">Here is a hypothetical example of RNA-Seq dataset from the drought experiment with hybrid maize, i.e. drought stress vs optimal water supply. Samples were sequenced with short reads being mapped onto the maize reference genome and the number of reads mapped to the features of interest (genic regions) were obtained. The gene expression profile for a particular sample consists of the set of gene-wise counts of sequence reads from that sample and one of the experimental objectives is to identify the differentially expressed genes.</p>
					<p>As the distribution of direct count data is highly skewed, a log2 transformation can approximately normalize the distributions as indicated below:</p>
					<p>
					<img src="https://latex.codecogs.com/gif.latex?t&space;=&space;log{_{2}}^{(C&space;&plus;&space;C_{0})}" title="t = log{_{2}}^{(C + C_{0})}" />
					<br>
					where, <em>C</em> is the count data while <img src="https://latex.codecogs.com/gif.latex?C_{0}" title="C_{0}" /> is a small positive constant (â€œ<em>1</em>", for instance) to facilitate the computation of the logarithm value in case of zero count for some conditions.
					</p>
					<p>The following table lists the top 12 rows of transformed data from the drought experiments:
					<img src="rimg/rimg10_data.png" alt="datatable" align="middle" width="100%">
					</p>
					<p id="rnaMAplot">The data set should be performed with additional manipulations to filter and exclude the genes with non-differential expression. A simple scatter plot can be generated using the script below prior to the differential gene expression analysis.
					<pre><code>
					ggplot(data=mydata, aes(x = CAW, y = CAD, color=CBD)) +
					  geom_point(size=3) +
					  labs(title="Scatterplot of log2 transformed gene expression data \nfrom the drought experiment of hybrid corn", x= "Log2 gene expression for CAW", y="Log2 gene expression for CAD") +
					  theme(
					    plot.title = element_text(color="#0000ff", hjust = 0.5, size=13, face="bold"),
					    axis.title.x = element_text(color="#003366", size=12, face="bold"),
					    axis.title.y = element_text(color="#003366", size=12, face="bold"))
					</code></pre>
					</p>

					<p>Then, an MA plot for differential expression analysis can be created to visualize the differences between measurements obtained from two samples or treatments based on the log-intensity ratio (<em>M</em>) and the average log-expression (<em>A</em>).
					</p>
					<img src="https://latex.codecogs.com/gif.latex?M&space;=&space;S_{2}&space;-&space;S_{1}" title="M = S_{2} - S_{1}" /> <br>
					<img src="https://latex.codecogs.com/gif.latex?A&space;=&space;\frac{S_{1}&space;&plus;&space;S_{2}}{2}" title="A = \frac{S_{1} + S_{2}}{2}" /> <br>
					where, <img src="https://latex.codecogs.com/gif.latex?S_{1}" title="S_{1}" /> and <img src="https://latex.codecogs.com/gif.latex?S_{2}" title="S_{2}" /> are the expression data for sample <em>1</em> and <em>2</em>, respectively.
					<br><br>
					<pre><code>
					ggplot(data=mydata, aes(x = A, y = M, color=CBD)) +
				         geom_point(size=3) +
				         labs(title="MA-plot for log2 transformed gene expression data \nfrom the drought experiment of hybrid corn", x= "Log2 mean expression", y="Log2 fold change\n(drought condition vs optimal water supply)") +
				         theme(
				         plot.title = element_text(color="#0000ff", hjust = 0.5, size=13, face="bold"),
				         axis.title.x = element_text(color="#003366", size=12, face="bold"),
				         axis.title.y = element_text(color="#003366", size=12, face="bold"))
					</code></pre>

					<br>
					To generate a heatmap for clustering gene expression leveles across samples and treatments:
					<br>
					<pre><code>
					library(stats)
					library(gplots)
					heatmap.2(as.matrix(mydata), col=redgreen(95), scale="row", key=T, keysize=1.5,density.info="none",
					      trace="none", cexCol=0.9, labRow=NA)
					</code></pre>

					<br>
					<p>
					The transformed dataset can be normalized using the following approach (minmax normalization):
					</p>
					<img src="https://latex.codecogs.com/gif.latex?x_{i}^{'}&space;=&space;\frac{x_{i}-min(x)}{max(x)-min(x)}" title="x_{i}^{'} = \frac{x_{i}-min(x)}{max(x)-min(x)}" />
					<br>
					where, <img src="https://latex.codecogs.com/gif.latex?x_{i}" title="x_{i}" /> and <img src="https://latex.codecogs.com/gif.latex?x_{i}^{'}" title="x_{i}^{'}" /> are the <img src="https://latex.codecogs.com/gif.latex?i^{th}" title="i^{th}" /> data before and after normalization, respectively.

					<br>
					<p>Thus, a heatmap of similarities can also be created among the participating genotypes based on the expressed genes under different treatment conditions:</p>
					<pre><code>
					library(RColorBrewer)
					mydata = as.matrix(dist(t(mydata)))
					mydata = (mydata - min(mydata))/(max(mydata) - min(mydata))
					hmcol = colorRampPalette(brewer.pal(9, "GnBu"))(100)
					heatmap(mydata, col=hmcol)
					</code></pre>

				</article>

				<aside id="sidebar">
					<br>
					<div class="purple">
					<a href="rimg/rimg11_scatterplot.png" style="text-decoration:none;">
					<img src="rimg/rimg11_scatterplot.png" alt="scatterplot" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">A scatter plot is created using log2 transformed gene expression data from two samples.</font></p>
					</a>
					</div>
					<br style ="line-height:6;">

					<div class="purple">
					<a href="rimg/rimg12_MAplot.png" style="text-decoration:none;">
					<img src="rimg/rimg12_MAplot.png" alt="MAplot" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">MA plot to visualize differential gene expression between samples or treatments.</font></p>
					</a>
					</div>
					<br style ="line-height:6;">

					<div class="purple">
					<a href="rimg/rimg13_hmgenexp.png" style="text-decoration:none;">
					<img src="rimg/rimg13_hmgenexp.png" alt="cim" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">A heatmap for clustering gene expression levels across samples and treatments.</font></p>
					</a>
					</div>
					<br style ="line-height:6;">

					<div class="purple">
					<a href="rimg/rimg14_cim.png" style="text-decoration:none;">
					<img src="rimg/rimg14_cim.png" alt="cim" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">A heatmap of similarities across the samples and treatments.</font></p>
					</a>
					</div>
				</aside>
			</div>

			<div class="container">
				<article id="main-col">
					<br>
					<h3 id="inbredhybrid">Chord diagram: an elite inbred and its multiple hybrids</h3>
					<p>Multiple high-yielding commercial hybrids produced from an elite inbred:</p>
					<pre><code>
					#corn inbred and yield
					library(circlize)
					mydata=read.table("...mydata.csv", sep=",")
					colnames(mydata)=c("FIL0a",	"SYT1",	"SYT2",	"SYT3")
					rownames(mydata)=c("FIL0a",	"SYT1",	"SYT2",	"SYT3")
					mat=data.matrix(mydata)
					chordDiagram(mat)
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="rimg/rimg7.png" style="text-decoration:none;">
					<img src="rimg/rimg7.png" alt="inbredhybrid" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">The elite inbred and its multiple commercial hybrids.</font></p>
					</a>
					</div>
				</aside>
			</div>

			<div class="container">
				<article id="main-col">
					<h3 id="chorddiagram">Chord diagram: the SNF genic contributions to phenotypes</h3>
					<p>The chord diagram from R is created for the genic contributions to phenotypic traits, such as biomass, protein or yield, from the main orthologs of the symbiotic nitrogen fixation (SNF) based on the averages of their gene expression levels.</p>
					Assumed that line1 (L1) contains the gene <em>Medtr1g062620</em>, L2 contains <em>Medtr3g077720</em>, L3 contains <em>Medtr5g014820</em>, L4 contains <em>Medtr1g062620</em> and <em>Medtr3g077720</em>, L5 contains <em>Medtr1g062620</em> and <em>Medtr5g014820</em>, L6 contains <em>Medtr3g077720</em> and <em>Medtr5g014820</em>, L7 contains <em>Medtr1g062620</em>, <em>Medtr3g077720</em> and <em>Medtr5g014820</em>.<br>
					<pre><code>
					#Chord diagram for the hypothetical contribution to yield from the orthologs responsible for symbiotic nitrogen fixation (SNF)
					library(circlize)
					mydata=read.table("...mydata.csv", sep=",")
					colnames(mydata)=c("L1", "L2",	"L3",	"L4",	"L5",	"L6",	"L7")
					rownames(mydata)=c("L1", "L2",	"L3",	"L4",	"L5",	"L6",	"L7")
					mat=data.matrix(mydata)
					chordDiagram(mat)
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="rimg/rimg6.jpg" style="text-decoration:none;">
					<img src="rimg/rimg6.jpg" alt="snfchord" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">The chord diagram for the SNF genic contributions.</font></p>
					</a>
					</div>
				</aside>
			</div>

			<div class="container">
				<article id="main-col">
					<h3 id="corrmatrix">Graphical display of a correlation matrix of soybean traits</h1>
					<pre><code>
					library(tidyverse)
					library(data.table)
					library(corrplot)
					soydata<- read.csv("sdata.csv")
					trtdata <- select(soydata, yield, disRes, height, lodging, seedwt, protein, oil, sumpo)
					matrix <- cor(trtdata);
					corrplot(matrix, method="color", type="upper")
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="rimg/rimg3.png" style="text-decoration:none;">
					<img src="rimg/rimg3.png" alt="simple plotting" height="100%" width="100%">
					<br>
					<p><font color="#ffffff">A correlation matrix of soybean traits.</font></p>
					</a>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h3 id="rregression">Creating regression charts between crop traits</h1>
					<pre><code>
					library(ggplot2)
					library(gridExtra)
					c1a <- ggplot(soydata, aes(x = disRes, y = yield)) +
								geom_point(aes(size=3, color="red") + geom_smooth(method = "lm", se = TRUE))
					c1b <- c1a + theme(axis.title.x = element_text(face="bold", size=12)) + labs(x="Disease
								Rating") + theme(axis.title.y = element_text(face="bold", size=12)) + labs(y="Yield") + theme(plot.title = element_text(face="bold", size=12)) + labs(title="Yield Response to Root Disease") + theme(panel.grid.major = element_blank(), panel.border = element_blank(), panel.background = element_blank(), axis.line = element_line(size=.4))
					m1 <- lm(soydata$yield ~ soydata$disRes)
					a1 <- signif(coef(m1)[1], digits = 2)
					b1 <- signif(coef(m1)[2], digits = 2)
					regeq1 <- paste("y = ",b1,"x + ",a1, sep="")
					regression1 <- c1b + geom_text(aes(x = 6, y = 360, label = regeq1), color="black", size=5,
						parse = FALSE)
					grid.arrange(r1, r2, nrow=1)
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<a href="rimg/rimg5.png" style="text-decoration:none;">
						<img src="rimg/rimg5.png" alt="simple plotting" height="100%" width="100%">
						<br>
						<p><font color="#ffffff">Regression figures between crop traits.</font></p>
						</a>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h3 id="scatterplotmatrices">Scatterplot matrices for soybean traits by correlation</h1>
					<pre><code>
					library(gclus)
					trtdata.r <- abs(cor(trtdata))
					trtdata.col <- dmat.color(trtdata.r)
					trtdata.o <- order.single(trtdata.r)
					cpairs(trtdata, trtdata.o, panel.colors=trtdata.col, gap=0.3,
					       main="Scatterplot Matrices for Soybean Traits by Correlation")
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
					<a href="rimg/rimg4.png" style="text-decoration:none;">
					<img src="rimg/rimg4.png" alt="Scatterplot" height="100%" width="100%">
					</a>
					</div>
				</aside>
			</div>
		</section>

		<section id="main">
			<div class="container">
				<article id="main-col">
					<h3>Simple plotting</h1>
					<pre><code>
					(A)
x<-1:500
y<-sin(x/10) * exp(x * -0.01)
plot(x, y, col="#308866") #plot the dependent variable vs the independent variable.
title(main="Simple plotting (A)", col.main="#000080")
(B)
x<-1:500
y<-sin(x/80) * exp(x * -0.001)
plot(x, y, col="#308866") #plot the dependent variable vs the independent variable.
title(main="Simple plotting (B)", col.main="#000080")
					</code></pre>
				</article>

				<aside id="sidebar">
					<div class="purple">
						<a href="rimg/rimg1.jpg" style="text-decoration:none;">
						<img src="rimg/rimg1.jpg" alt="simple plotting" height="80%" width="100%">
						</a>
					</div>
				</aside>
			</div>
		</section>
		<br><br>

		<footer>
			<p>Data Science, Copyright &copy; 2020
			<a href="https://twitter.com/ChunlinHe" target="_blank"><img src="img/tweet_icon.png" width="31" height="31" alt="" /></a>
			<a href="https://www.linkedin.com/in/chunlinhe888" target="_blank"><img src="img/in_icon.png" width="31" height="31" alt="" /></a>
			</p>
			<p id="time"></p>
			<script>
			document.getElementById("time").innerHTML = Date();
			</script>
		</footer>
</body>

</html>
